File: rest-api/src/main/java/com/fuelstation/managmentapi/fuelstation/domain/models/FuelTank.java
package com.fuelstation.managmentapi.fuelstation.domain.models;

import com.fuelstation.managmentapi.common.domain.FuelGrade;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.time.LocalDate;
import java.util.Optional;

@Data
@AllArgsConstructor
public class FuelTank{
    private Long id;
    private FuelGrade fuelGrade;
    private float currentVolume;
    private float maxCapacity;
    private Optional<LocalDate> lastRefilDate;

    public float getAvailableVolume() {
        return this.maxCapacity - this.currentVolume;
    }
}

File: rest-api/src/main/kotlin/com/fuelstation/managmentapi/fuelstation/domain/models/FuelTank.kt
package com.fuelstation.managmentapi.fuelstation.domain.models

import com.fuelstation.managmentapi.common.domain.FuelGrade
import java.time.LocalDate


data class FuelTank(
    val id: Long, 
    val fuelGrade: FuelGrade,
    val currentVolume: Float,
    val maxCapacity: Int,
    val lastRefilDate: LocalDate
)